src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -229,25 +232,96 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(Runti
         /// have GraphQL configuration: when entity.GraphQL == false or null.
         /// </summary>
         /// <seealso cref="https://spec.graphql.org/October2021/#Name"/>
-        /// <param name="entityCollection">The runtime entities to process.</param>
-        public static void ValidateEntityNamesInConfig(RuntimeEntities entityCollection)
+        /// <param name="runtimeConfig">The runtime configuration.</param>
+        public static void ValidateEntityConfiguration(RuntimeConfig runtimeConfig)
         {
-            foreach ((string _, Entity entity) in entityCollection)
+            // Stores the unique rest paths configured for different entities present in the config.
+            HashSet<string> restPathsForEntities = new();
+
+            foreach ((string entityName, Entity entity) in runtimeConfig.Entities)
             {
-                if (entity.GraphQL.Enabled)
+                if (runtimeConfig.Runtime.Rest.Enabled)
+                {
+                    // By default we assume Rest is enabled for the entity.
+                    bool isRestEnabledForEntity = true;
+
+                    // If no custom rest path is defined for the entity, we default it to the entityName.
+                    string pathForEntity = entityName;
+                    if (entity.Rest is not null)
+                    {
+                        pathForEntity = entity.Rest.Path is not null ? entity.Rest.Path.TrimStart('/') : entityName;
+                        isRestEnabledForEntity = entity.Rest.Enabled;
+                        SupportedHttpVerb[] supportedOperations = entity.Rest.Methods;
+
+                        // Since 'path' is an optional property, we skip validation if its absent.
+                        ValidateAndGetRestSettingsForEntity(entityName, pathForEntity);
+                    }
+
+                    if (isRestEnabledForEntity && !restPathsForEntities.Add(pathForEntity))
+                    {
+                        // Presence of multiple entities having the same rest path configured causes conflict.
+                        throw new DataApiBuilderException(
+                            message: $"The rest path: {pathForEntity} specified for entity: {entityName} is already used by another entity.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                            );
+                    }
+                }
+
+                // If GraphQL endpoint is disabled globally, we skip the validations related to it.
+                if (!runtimeConfig.Runtime.GraphQL.Enabled)
+                {
+                    continue;
+                }
+
+                if (entity.GraphQL is null)
+                {
+                    continue;
+                }
+                else if (entity.GraphQL.Enabled)
```

Comment:
instead of doing 2 continues, can we simplify this into a single if statement?

`if (runtimeConfig.Runtime.GraphQL.Enabled && entity.GraphQL is not null && entity.GraphQL.Enabled)` ?

Reply:
Yeah we can. A correction here, the condition would be:

`if (!runtimeConfig.Runtime.GraphQL.Enabled || entity.GraphQL is null || !entity.GraphQL.Enabled)`